qml/pages/GalleryUI.qml

```diff
@@ -0,0 +1,59 @@
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.Media 1.0
+import QtMultimedia 5.4
+import QtQuick.Layouts 1.0
+import uk.co.piggz.harbour_advanced_camera 1.0
+import Nemo.Configuration 1.0
+import "../components/"
+
+Page {
+    id: galleryPage
+
+    property var photoList: ({})
+
+    // The effective value will be restricted by ApplicationWindow.allowedOrientations
+    allowedOrientations: Orientation.Landscape
+
+    SlideshowView {
+        id: gallery
+
+        clip: true
+        width: parent.width
+        height: parent.height
+        itemWidth: width / 2
+        itemHeight: height * 2 / 3
+        y: parent.height / 6
+
+        model: photoList
+
+        delegate: Rectangle {
+            id: delegate
+            width: gallery.itemWidth
+            height: gallery.height
+            border.width: 1
+            color: 'black'
+
+            Image {
+                id: thumbnail
+
+                asynchronous: true
+                sourceSize.width: parent.width
+                anchors.fill: parent
+                fillMode: Image.PreserveAspectFit
+                source: photoPath
+
+                MouseArea {
+                    anchors.fill: parent
+                    onClicked: {
+                        console.log("Clicked", thumbnail.source)
+                        Qt.openUrlExternally(thumbnail.source)
+                    }
+                }
+            }
+
+        }
+    }
+
+
```

Comment:
Cleanup useless empty rows. Between items these can be used for readability but no need to have these here and some other places.